### PR TITLE
Add custom function and property support

### PR DESCRIPTION
**Pull request by HayrapetyanMurad**
Add support of adding custom lua c function as a class method.
Add support of adding global custom lua c functions.
Add support of adding custom lua c function as a class property. 

- - -
This change is [Reviewable](https://reviewable.io/reviews/PicsArt/luabind/5)

### CODE REVIEW

File: tests/custom_functions.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+public:
+    IntWrapper(int value)
+        : _value(value) {}
+
+    int get() const {
+        return _value;
+    }
+
+public:
+    static int toLuaTable(lua_State* state) {
```
Comment (davits) [inline tests/custom_functions.cpp]: Lets follow the overall style of the project and name the lua_State* variable `L`

File: tests/explicit_delete.cpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        luabind::class_<Deletable>(L, "Deletable").constructor<>("new").construct_shared<>("makeShared");
+
+        EXPECT_EQ(lua_gettop(L), 0);
+    }
+};
+
+TEST_F(ExplicitDeleteTest, ExplicitDelete) {
+    run(R"--(
+        obj = Deletable:new()
+        obj:delete()
```
Comment (davits) [inline tests/explicit_delete.cpp:28]: Make sure that double deletion, either manually or from gc will pose a problem.

**Review by davits — COMMENTED** [review]
Looks good, just minor improvements needed.

File: tests/explicit_delete.cpp (hunk 2 above)
Comment (HayrapetyanMurad) [reply tests/explicit_delete.cpp:28]: Sorry didn't comprehend what you mean. If we call delete() on the same object twice the behavior is defined, first delete will call the user_data destructor which will delete the data, the second will not call user_data destructor as object stored in user_data is already null.

File: tests/custom_functions.cpp (hunk 1 above)
Comment (HayrapetyanMurad) [reply tests/custom_functions.cpp]: Done.

File: tests/explicit_delete.cpp (hunk 2 above)
Comment (davits) [reply tests/explicit_delete.cpp:28]: Sorry, meant to write "will not pose a problem". 
I thing this test should cover that:
``` lua
obj = Deletable:new()
obj:delete()
obj:delete()
obj = null
collectgarbage()
```

Comment (HayrapetyanMurad) [reply tests/explicit_delete.cpp:28]: Done.

**Review by davits — APPROVED** [review]

**Pull request merged**